BlueShift-iOS-SDK/BlueShiftAppDelegate.m

```diff
@@ -1004,10 +1004,30 @@ - (void)trackAlertDismiss {
     [[BlueShift sharedInstance] trackEventForEventName:kEventDismissAlert andParameters:nil canBatchThisEvent:YES];
 }
 
-/// SDK triggeres app_open event automatically on app launch and is controlled by the enableAppOpenTrackEvent  config flag
+/// SDK triggeres app_open event automatically when app is launched from killed state and is controlled by the enableAppOpenTrackEvent config flag
+/// @discussion The automatic app_open events can be throttled by setting time interval in secods to config.automaticAppOpenTimeInterval.
 - (void)trackAppOpenOnAppLaunch:(NSDictionary *)parameters {
-    if ([BlueShift sharedInstance].config.enableAppOpenTrackEvent) {
-        [self trackAppOpenWithParameters:parameters];
+    @try {
+        if ([BlueShift sharedInstance].config.enableAppOpenTrackEvent) {
+            if ([BlueShift sharedInstance].config.automaticAppOpenTimeInterval == 0) {
+                [self trackAppOpenWithParameters:parameters];
+            } else {
+                double lastAppOpenTimestamp = [[NSUserDefaults standardUserDefaults] doubleForKey:kBlueshiftLastAppOpenTimestamp];
+                double nowTimestamp = [[NSDate date] timeIntervalSince1970];
+                if (lastAppOpenTimestamp != 0) {
```

Comment:
suggestion: if we can move the logic for checking if we should fire an app_open to a method (say; shouldAcceptAppOpen or something) and keep this method clean, it would be more readable.

Reply:
Makes sense, I have added new helper method for the same.